packages/api/src/user.js

```diff
@@ -347,6 +300,72 @@ export async function userUploadsRename (request, env) {
   return new JSONResponse(res)
 }
 
+/**
+ * List a users pins regardless of the token used.
+ * As we don't want to scope the Pinning Service API to users
+ * we need a new endpoint as an umbrella.
+ *
+ * @param {AuthenticatedRequest} request
+ * @param {import('./env').Env} env
+ */
+export async function userPinsGet (request, env) {
+  const requestUrl = new URL(request.url)
+  const { searchParams } = requestUrl
+
+  const { size, page, offset, before, after, sortBy, sortOrder } = pagination(searchParams)
+  const urlParams = new URLSearchParams(requestUrl.search)
+  const params = Object.fromEntries(urlParams)
+
+  const psaParams = validateSearchParams(params)
+  if (psaParams.error) {
+    throw psaParams.error
+  }
+
+  const tokens = (await env.db.listKeys(request.auth.user._id)).map((key) => key._id)
+
+  let pinRequests
+
+  try {
+    pinRequests = await env.db.listPsaPinRequests(tokens, {
```

Comment:
Would it make more sense here to update the `listPsaPinRequests` method to optionally take a user ID instead of a list of tokens, and have it do the JOIN in the DB, rather than doing two separate queries in sequence?

Reply:
Yeah there is definitely an argument for that here.
The reason I chose against that approach is because
A) we can reuse exactly the same query (and therefore method) as the Pinning Service API by just passing tokens
B) we don't want to give the PSA awareness of user management

But you are absolutely right, this is more computationally expensive than doing it in the DB. It just seemed the cleanest approach at the time. I'll take your lead on what you reckon.